mmv1/templates/terraform/pre_create/alloydb_cluster.go.erb

```diff
@@ -63,3 +62,20 @@ if secondaryConfig != nil {
 		return fmt.Errorf("Error creating cluster. Add {cluster_type: \"SECONDARY\"} if attempting to create a secondary cluster, otherwise remove the secondary_config.")
 	}
 }
+
+// Read the maintenance update policy config from obj and validate the input variables
+var maintenanceUpdatePolicy map[string]interface{}
+if val, ok := obj["maintenanceUpdatePolicy"]; ok {
+	maintenanceUpdatePolicy = val.(map[string]interface{})
+
+	// Validate maintenance window variables
+	if maintenanceUpdatePolicy["maintenanceWindows"] != nil {
+		mws := maintenanceUpdatePolicy["maintenanceWindows"].([]interface{})
+		for _, mw := range mws {
+			maintenanceWindow := mw.(map[string]interface{})
+			if maintenanceWindow["day"] == nil || maintenanceWindow["startTime"] == nil {
+				return fmt.Errorf("Error creating cluster. Both \"day\" and \"startTime\" are required if attempting to configure maintenance windows, otherwiese remove the maintenance_windows.")
```

Comment:
This can be represented by `required: true` in the YAML file for these fields. Then this can be removed

Reply:
so I just set the required at day and start_time parameter? and this will not make the entire parent nested object required?

Reply:
required only applies within the sub-object, so it means if the parent object is specified then `day` and `start_time` are required.

Marking the parent object as required is the only way to require that it exists at all